op-node/p2p/sync.go

```diff
@@ -209,12 +208,15 @@ type SyncClient struct {
 	quarantineByNum map[uint64]common.Hash
 
 	// inFlight requests are not repeated
-	inFlight map[uint64]*atomic.Bool
-
-	requests       chan rangeRequest
-	peerRequests   chan peerRequest
+	inFlight       map[uint64]bool
```

Comment:
Since mapping to a `atomic.Bool` does not provide concurrency safety against deleting that value from the mapping, this was giving a false sense of security. The code now deletes a value (instead of setting its value to `false`) anytime it wants to flag that a blockNum is no longer `inFlight`, so I think the vanilla `bool` is a better fit.

Reply:
The problem is that this mapping is used from both `mainLoop` and `peerLoop` so it seems to me that it needs to be thread safe.  It used to be safe because the `atomic.Bool` was all that was shared between the `mainLoop` and `peerLoop` with the `peerLoop` just setting it to false and only the `mainLoop` would ever modify the map.

So I think we've introduced a concurrency issue with this change.

Reply:
Good catch. Added mutex for `inFlight` map in https://github.com/ethereum-optimism/optimism/pull/10063/commits/1feada65b665ed2348dbce71b3eb3af7f5deed9b